### PR TITLE
WEEK12

### PR DESCRIPTION
**Pull request by KYM-P**
Week12 Assignment Created

* 실습 내용과 동일
 > * Bottom Navigation 사용
 >> 무지개, 숫자, 알파벳 3가지의 메뉴, 선택된 item 마다 viewPager2 Adapter 변경
 > * ViewPager2 사용

* 실습 외
 > * TabLayout 사용
 >> 무지개, 숫자, 알파벳 3가지의 메뉴마다 달라지는 viewPager2 에 맞춰 tabLayout 사용
 > * ViewPager2 에니매이션 추가

### CODE REVIEW

File: WEEK12/app/src/main/java/com/example/week12/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        // numberFragment
+        val numberFragments = mutableListOf<Fragment>()
+        numberList.forEach { i ->
+            numberFragments.add(NumberViewFragment(i.toString()))
+        }
+        // alphabetFragment
+        val alphabetFragments = mutableListOf<Fragment>()
+        alphaberList.forEach { c ->
+            alphabetFragments.add(AlphabetViewFragment(c.toString()))
+        }
```
Comment (skdud0629) [inline WEEK12/app/src/main/java/com/example/week12/MainActivity.kt:42]:
```
fun <T> createFragments(list: List<T>, fragmentCreator: (T) -> Fragment): MutableList<Fragment> {
    return list.map(fragmentCreator).toMutableList()
}

```

반복 코드는 함수로 묶어주세요!

File: WEEK12/app/src/main/java/com/example/week12/MainActivity.kt [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            }
+            return@setOnItemSelectedListener true
+        }
+    }
+
+    fun<T> setViewPager(fragments : MutableList<Fragment>, list : List<T> ) {
+        binding.viewPager2.adapter = ViewPager2Adapter(this, fragments)
+        TabLayoutMediator(binding.tabLayout,binding.viewPager2) { tab, position ->
+            tab.text = list[position].toString()
+        }.attach()
```
Comment (skdud0629) [inline WEEK12/app/src/main/java/com/example/week12/MainActivity.kt:72]: tabLayoutMediator?.detach()
요녀석에 대해서도 알아보셨으면 좋으실 것 같아요! 효율적인 메모리 관리를 위해!

File: WEEK12/app/src/main/java/com/example/week12/viewpager2/ZoomOutPageTransformer.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            val pageWidth = width
+            val pageHeight = height
+            when {
+                position < -1 -> { // [-Infinity,-1)
+                    // This page is way off-screen to the left.
+                    alpha = 0f
+                }
+                position <= 1 -> { // [-1,1]
+                    // Modify the default slide transition to shrink the page as well.
+                    val scaleFactor = Math.max(MIN_SCALE, 1 - Math.abs(position))
```
Comment (skdud0629) [inline WEEK12/app/src/main/java/com/example/week12/viewpager2/ZoomOutPageTransformer.kt:24]: Math.max() 대신 Kotlin에서 지원하는 kotlin.math.max()를 사용해주세용

**Review by skdud0629 — COMMENTED** [review]
수고하셨어요!
